idx/memory/memory.go

```diff
@@ -242,17 +237,17 @@ func (m *MemoryIdx) add(def *schema.MetricDefinition) error {
 		log.Debug("memory-idx: creating branch %s with child %s", branch, nodes[i])
```

Comment:
startPos is used after this loop and I dont see how i can maintain the same logic when removing `i`